### PR TITLE
Improve safelinks pattern

### PR DESCRIPTION
**Pull request by jan-kiszka**
This also ensure that the complete link is replaced when citing a
mangled email.

### CODE REVIEW

**Comment by jan-kiszka** [comment]
BTW, how about releasing a new version that contains link-unmangling for replies?

File: src/decoders.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-let outlook_regex = /https:\/\/\w+\.safelinks\.protection\.outlook\.com\/\?url=(\S+?)((?:\&amp;data.*)+$|$)/i;
+let outlook_regex = /https:\/\/\w+\.safelinks\.protection\.outlook\.com\/\?url=([^&]*)(&[-+a-zA-Z0-9%&/=.;]*)?/i;
```
Comment (cphyc) [inline src/decoders.js]: Shouldn't the `+`, `-` and `.` be escaped above?
```suggestion
let outlook_regex = /https:\/\/\w+\.safelinks\.protection\.outlook\.com\/\?url=([^&]*)(&[\-\+a-zA-Z0-9%&/=\.;]*)?/i;
```

**Review by cphyc — APPROVED** [review]
Thanks for the PR!
Would it be possible to add a test for this case?

Once merged, I'll happily create a new release.

File: src/decoders.js (hunk 1 above)
Comment (jan-kiszka) [reply src/decoders.js]: Don't ask me when exactly they need to be escaped. This seems to work fine. But there is more to improve, e.g. excluding trailing `.`.

**Comment by jan-kiszka** [comment]
Will have a look into test cases.

**Comment by jan-kiszka** [comment]
Further tuned the regexps and also added some tests. Seems to work fine here so far.

**Comment by cphyc** [comment]
Thanks for the PR! I also created a new release that includes your code. You should be able to update from TB directly.

**Pull request merged**